### PR TITLE
Data Documentation Guide improvements

### PR DESCRIPTION
**Pull request by eugmandel**
1. Clarified that data documentation is configurable and the user has the control.
2. Documented the run_id_filter and the stores configuration

### CODE REVIEW

File: docs/guides/data_documentation.rst [hunk 1]
```diff
@@ -105,12 +108,20 @@ Here is an example of a site configuration:
                 module: great_expectations.render.view
                 class: DefaultJinjaPageView
 
+* ``validations_store`` and ``profiling_store`` in the example above specify the location of validation and profiling results that the site will include in the documentation. The store's ``type`` can be ``filesystem`` or ``s3``. S3 store is not currently implemented, but will be supported in a near future. ``base_directory`` must be specified for ``filesystem`` stores. The optional ``run_id_filter`` attribute allows to include (``eq`` for exact match) or exclude (``ne``) validation results with a particular run id.
```
Comment (jcampbell) [inline docs/guides/data_documentation.rst]: S3 store is not currently implemented, but will be supported in a near future.
->
(S3 store is not currently implemented, but will be supported in the near future.)

File: docs/guides/data_documentation.rst [hunk 2]
```diff
@@ -105,12 +108,20 @@ Here is an example of a site configuration:
...
+
+
+
+Adjusting Data Documentation For Your Project's Needs
+*****************************************************
 
 By default, GE creates two data documentation sites for a new project:
 
 1. "local_site" renders documentation for all the datasources in the project from GE artifacts in the local repo. The site includes expectation suites and profiling and validation results from `uncommitted` directory. Local site provides the convenience of visualizing all the entities stored in JSON files as HTML.
```
Comment (jcampbell) [inline docs/guides/data_documentation.rst]: ...results from `uncommitted` directory.
->
...results from the `uncommitted` directory.

Comment (jcampbell) [inline docs/guides/data_documentation.rst]: in the local repo.
->
in the local filesystem.

File: docs/guides/data_documentation.rst [hunk 3]
```diff
@@ -105,12 +108,20 @@ Here is an example of a site configuration:
...
 2. "team_site" is meant to support the "shared source of truth for a team" use case. By default only the expectations section is enabled. Users have to configure the profiling and the validations sections (and the corresponding validations_store and profiling_store attributes based on the team's decisions where these are stored (a local filesystem or S3). Reach out on `Slack <https://tinyurl.com/great-expectations-slack>`__ if you would like to discuss the best way to configure a team site.
```
Comment (jcampbell) [inline docs/guides/data_documentation.rst]: (and the corresponding validations_store and profiling_store attributes based on the team's decisions where these are stored (a local filesystem or S3).
->
(and the corresponding validations_store and profiling_store attributes) based on the team's decisions about where these are stored.

File: docs/guides/data_documentation.rst [hunk 4]
```diff
@@ -52,7 +52,10 @@ Users can specify
 * where the HTML files should be written (filesystem or S3)
 * which renderer and view class should be used to render each section
 
-Here is an example of a site configuration:
```
Comment (jcampbell) [inline docs/guides/data_documentation.rst:55]: There are three use cases for using documentation in a data project:
->
The GE DataContext uses a configurable "data documentation site" to define which artifacts to compile and how to render them as documentation. Multiple sites can be configured inside a project, each suitable for a particular data documentation use case.

For example, we have identified three common use cases for using documentation in a data project. They are to:

1. Visualize all Great Expectations artifacts in the local repo of a project as HTML: expectation suites, validation results and profiling results.
2. Maintain a "shared source of truth" for a team working on a data project. This documentation renders all the artifacts committed in the source control system (expectation suites and profiling results) and a continuously updating data quality report, built from a chronological list of validations by run id.
3. Share a spec of a dataset with a client or a partner. This is similar to API documentation in software development. This documentation would include profiling results of the dataset to give the reader a quick way to grasp what the data looks like, and one or more expectation suites that encode what is expected from the data to be considered valid.

**Review by jcampbell — CHANGES_REQUESTED** [review]
Some line-in/line-out changes targeting the concept introduction order we talked about briefly on Monday

File: docs/guides/data_documentation.rst (hunk 1 above)
Comment (eugmandel) [reply docs/guides/data_documentation.rst]: implemented

Comment (eugmandel) [reply docs/guides/data_documentation.rst]: implemented

File: docs/guides/data_documentation.rst (hunk 2 above)
Comment (eugmandel) [reply docs/guides/data_documentation.rst]: implemented

Comment (eugmandel) [reply docs/guides/data_documentation.rst]: implemented

**Review by jcampbell — APPROVED** [review]

**Pull request merged**